### PR TITLE
[jenkins]updated version of OAI Jenkins pipeline

### PR DESCRIPTION
**Pull request by 119Vik**
Signed-off-by: Vitalii <vitalij.kostenko@gmail.com>

## Summary

Updated OAI Jenkinsfile according to switching to GitHub and using private credentials

### CODE REVIEW

File: ci-scripts/JenkinsFile-GitLab [hunk 1]
```diff
@@ -83,45 +80,53 @@ pipeline {
...
-
-          sh "tar -cjhf /tmp/converged_mme.tar.bz2 .git"
-          sh "mv /tmp/converged_mme.tar.bz2 ."
-          copyTo2ndServer('converged_mme.tar.bz2', '.', new_host_flag, new_host_user, new_host)
-          myShCmd('git checkout -f ' + TEMP_COMMIT, new_host_flag, new_host_user, new_host)
-          myShCmd('git clean -x -d -e .cache -e lte/gateway/.vagrant -f > /dev/null 2>&1', new_host_flag, new_host_user, new_host)
-          myShCmd('git status --ignored', new_host_flag, new_host_user, new_host)
-          sh "mkdir -p archives"
-          myShCmd('mkdir -p archives', new_host_flag, new_host_user, new_host)
+            // sh "git clone https://github.com/magma/magma.git ."
```
Comment (ssanadhya) [inline ci-scripts/JenkinsFile-GitLab]: Remove commented code

File: ci-scripts/JenkinsFile-GitLab [hunk 2]
```diff
@@ -83,45 +80,53 @@ pipeline {
...
+            // sh "git clone https://github.com/119Vik/magma-1.git ."
+            checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: '${sha1}']], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'vik_github', refspec: '+refs/pull/*:refs/remotes/origin/pr/*', url: 'https://github.com/magma/magma.git']]]    
+            GIT_COMMIT = sh returnStdout: true, script: 'git log -1 --pretty="%H"'
+            // echo "Git URL         is ${GIT_URL}"
```
Comment (ssanadhya) [inline ci-scripts/JenkinsFile-GitLab]: Remove commented code

File: ci-scripts/JenkinsFile-GitLab [hunk 3]
```diff
@@ -255,11 +262,11 @@ pipeline {
 
               // We install a dedicated version (installed on our CI server).
               myShCmd('docker cp /opt/clang-format/9.0.0/bin/clang-format ci-cn-clang-formatter:/usr/local/bin', new_host_flag, new_host_user, new_host)
-              if ("MERGE".equals(env.gitlabActionType)) {
-                myShCmd('docker exec -it ci-cn-clang-formatter /bin/bash -c "cd /home && ./ci-scripts/checkCodingFormattingRules.sh --src-branch ' + env.ghprbSourceBranch +' --target-branch ' + env.ghprbTargetBranch + '"', new_host_flag, new_host_user, new_host)
-              } else {
+              // if ("MERGE".equals(env.gitlabActionType)) {
```
Comment (ssanadhya) [inline ci-scripts/JenkinsFile-GitLab]: Remove commented code

**Review by ssanadhya — CHANGES_REQUESTED** [review]

File: ci-scripts/JenkinsFile-GitLab (hunk 1 above)
Comment (119Vik) [reply ci-scripts/JenkinsFile-GitLab]: fixed

File: ci-scripts/JenkinsFile-GitLab (hunk 3 above)
Comment (119Vik) [reply ci-scripts/JenkinsFile-GitLab]: fixed

**Comment by 119Vik** [comment]
ReRunOAITests

**Comment by 119Vik** [comment]
ReRunOAITests

File: ci-scripts/JenkinsFile-GitLab [hunk 4]
```diff
@@ -80,45 +77,30 @@ pipeline {
...
-
-          sh "tar -cjhf /tmp/converged_mme.tar.bz2 .git"
-          sh "mv /tmp/converged_mme.tar.bz2 ."
-          copyTo2ndServer('converged_mme.tar.bz2', '.', new_host_flag, new_host_user, new_host)
-          myShCmd('git checkout -f ' + TEMP_COMMIT, new_host_flag, new_host_user, new_host)
-          myShCmd('git clean -x -d -e .cache -e lte/gateway/.vagrant -f > /dev/null 2>&1', new_host_flag, new_host_user, new_host)
-          myShCmd('git status --ignored', new_host_flag, new_host_user, new_host)
-          sh "mkdir -p archives"
-          myShCmd('mkdir -p archives', new_host_flag, new_host_user, new_host)
+            checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: '${sha1}']], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'vik_github', refspec: '+refs/pull/*:refs/remotes/origin/pr/*', url: "https://{{GIT_URL}}.git"]]]
```
Comment (rdefosse) [inline ci-scripts/JenkinsFile-GitLab]: @119Vik 

2 things about this line.

-  Now that we have a public repository, do we still need to provide credentials? I don't think so.
-  I am not sure that you are handling the pull request correctly: my meaning is that we shall test the potential merge of the contributor's branch with the target branch. Not just the contributor's branch.
   *  I understand that the policy is to force the contributor to resync its branch against the latest version of  `master`.
   *  But then you need  a check to ensure it
   *  That is what I was doing in the section you removed.

Comment (119Vik) [reply ci-scripts/JenkinsFile-GitLab]: Both make sense for me - let me update my PR.

**Review by tmdzk — APPROVED** [review]

File: ci-scripts/JenkinsFile-GitLab [hunk 5]
```diff
@@ -280,39 +251,33 @@ pipeline {
...
           steps {
             script {
-              gitlabCommitStatus(name: "Activate-Test-VM") {
                 myShCmdWithLog('cd lte/gateway && vagrant destroy --force magma_test && vagrant up magma_test', 'archives/magma_vagrant_test_up.log', new_host_flag, new_host_user, new_host)
                 // The Traffic Testcases are disabled from the Mandatory suite --> will be run manually
                 myShCmd('sed -i -f ci-scripts/removeLastUlTestCases.sed lte/gateway/python/integ_tests/defs.mk', new_host_flag, new_host_user, new_host)
                 myShCmdWithLog('cd lte/gateway && vagrant ssh magma_test -c "cd magma/lte/gateway/python/ && make"', 'archives/magma_vagrant_test_make1.log', new_host_flag, new_host_user, new_host)
                 myShCmdWithLog('cd lte/gateway && vagrant ssh magma_test -c "cd magma/lte/gateway/python/integ_tests/ && make"', 'archives/magma_vagrant_test_make2.log', new_host_flag, new_host_user, new_host)
-              }
+              // }
```
Comment (ssanadhya) [inline ci-scripts/JenkinsFile-GitLab]: Remove commented code

**Review by ssanadhya — CHANGES_REQUESTED** [review]

File: ci-scripts/JenkinsFile-GitLab (hunk 5 above)
Comment (119Vik) [reply ci-scripts/JenkinsFile-GitLab]: thanks, good catch. fixed.

**Comment by 119Vik** [comment]
ReRunOAITests

**Comment by 119Vik** [comment]
ReRunOAITests

**Comment by 119Vik** [comment]
ReRunOAITests

**Comment by 119Vik** [comment]
ReRunOAITests

**Review by ssanadhya — APPROVED** [review]

**Pull request merged**